compiler-rt/lib/mpk_untrusted/alloc_site_handler.h

```diff
@@ -121,17 +123,37 @@ class AllocSite {
   // previous allocation sites for a given reallocated pointer.
   alloc_set_type& getAssociatedSet() { return associatedSet; }
 
+  // This struct is required for hashing the AllocSite properly in the unordered_set.
+  // Some Additional shuffling is used in the hashing of multiple values to ensure 
+  // that if the Function name and BasicBlock name happen to match, they do not 
+  // cancel each other out.
+  struct Hasher {
+    std::size_t operator()(const AllocSite& AC) const {
+      return ((std::hash<std::string>()(AC.getFuncName())
+              ^ (std::hash<std::string>()(AC.getBBName()) << 1)) >> 1)
+              ^ (std::hash<int64_t>()(AC.id()) << 1);
+    }
+  };
+
+  // Used for internal associated set.
   bool operator<(const AllocSite &ac) const {
```

Comment:
Is this still a thing? would have imagined using unordered_set would mean you'd remove this code.

Reply:
Removed due to commit `fea9c81`.